src/ucp/api/ucp.h

```diff
@@ -262,31 +262,34 @@ enum ucp_ep_params_field {
-                                                           @ref ucs_sock_addr_t
-                                                           sockaddr field
-                                                           must be provided and
-                                                           contain the address
-                                                           of the remote peer */
-    UCP_EP_PARAMS_FLAGS_NO_LOOPBACK    = UCS_BIT(1),  /**< Avoid connecting the
-                                                           endpoint to itself when
-                                                           connecting the endpoint
-                                                           to the same worker it
-                                                           was created on.
-                                                           Affects protocols which
-                                                           send to a particular
-                                                           remote endpoint, for
-                                                           example stream */
-    UCP_EP_PARAMS_FLAGS_SEND_CLIENT_ID = UCS_BIT(2)   /**< Send client id
-                                                           when connecting to remote
-                                                           socket address as part of the
-                                                           connection request payload.
-                                                           On the remote side value
-                                                           can be obtained from
-                                                           @ref ucp_conn_request_h using
-                                                           @ref ucp_conn_request_query */
+    /**
+     * Using a client-server connection establishment mechanism.
+     * @ref ucs_sock_addr_t sockaddr field must be provided and contain the
+     * address of the remote peer.
+     */
+    UCP_EP_PARAMS_FLAGS_CLIENT_SERVER  = UCS_BIT(0),
+
+    /**
+     * Avoid connecting the endpoint to itself when connecting the endpoint
+     * to the same worker it was created on. Affects protocols that send to
+     * a particular remote endpoint, for example, stream.
+     */
+    UCP_EP_PARAMS_FLAGS_NO_LOOPBACK    = UCS_BIT(1),
+
+    /**
+     * Send client identifier when connecting to remote socket address as
+     * part of the connection request payload. On the remote side the value
+     * can be obtained from @ref ucp_conn_request_h using
+     * @ref ucp_conn_request_query.
+     */
+    UCP_EP_PARAMS_FLAGS_SEND_CLIENT_ID = UCS_BIT(2),
+
+    /**
+     * Request selection of transports with capability to perform local
+     * operations using a peer's memory buffer and memory handle associated
+     * with it, and created with @ref UCP_MEM_MAP_EXPORTED capability.
+     */
+    UCP_EP_PARAMS_FLAGS_EXPORTED_MEMH  = UCS_BIT(3)
```

Comment:
why need it in ep params?

Reply:
to make sure transports that support XGVMI will be selected out of the box.
e.g. if two processes running on the same DPU, they will select SHM, but we need IB transports to perform  operations using imported memory key

Reply:
what if we always select such transports when running on a DPU?

Reply:
> what if we always select such transports when running on a DPU?

if only they are selected, we couldn't perform xgvmi

Reply:
I mean, what if when running on DPU we always make sure we also select lanes that support xgvmi? so even when both processes run of DPU, they will use it instead of SHM

Reply:
do you mean to somehow understand that we run on DPU/arm and select lanes which support xgvmi w/o requiring to pass `UCP_EP_PARAMS_FLAGS_EXPORTED_MEMH` flag?

Reply:
yes

Reply:
but it is not reasonable in the following use cases:
- two processes running on same DPU and perform operations w/o involving HOST memory - better to use SHM;
- two peers running on different DPUs and they have `UCX_CM_USE_ALLDEVICES=n` and no RDMACM can be used - so, UCP won't be able to use TCP.

Reply:
I take the view that it should be decided by user how a specific UCP endpoint is going to be used - for communication w/ DPU system memory (ne need to pass `UCP_EP_PARAMS_FLAGS_EXPORTED_MEMH`) or HOST system memory (need to pass `UCP_EP_PARAMS_FLAGS_EXPORTED_MEMH`).

Reply:
can we create both types of lanes (for DPU system memory and HOST memory) and use the right one for each operation according to operation parameters?

Reply:
I forgot to mention that `UCP_EP_PARAMS_FLAGS_EXPORTED_MEMH` flag is specified for HOST-side endpoints to make sure xgvmi-capable transports will be selected on HOST too.
So, we will need to select lanes according to their characteristics and then add more lanes with xgvmi capability  on HOST and DPU (if they not exist in initial selection).

Reply:
why is it needed on host side? i thought cross-gvmi could work even if you don't have any endpoint between host and DPU and you send the packed exported mkey by any OOB method,

Reply:
> why is it needed on host side? i thought cross-gvmi could work even if you don't have any endpoint between host and DPU and you send the packed exported mkey by any OOB method,

right, doesn't make sense to set this flag on HOST side.

if we decided to not have this flag, then we should somehow detect that process created on DPU/arm - and it should be some MD attribute or capability, right?